test/anvil/anvil-global-setup.ts

```diff
@@ -1,24 +1,71 @@
 import { CreateAnvilOptions, createAnvil } from '@viem/anvil';
 import { sleep } from '../lib/utils/promises';
 
+type NetworkSetup = {
+    rpcEnv: string;
+    fallBackRpc: string | undefined;
+    port: number;
+    forkBlockNumber: bigint;
+};
+
+type Networks = NetworkSetup[];
+
+const NETWORKS: Networks = [
+    {
+        rpcEnv: 'ETHEREUM_RPC_URL',
+        fallBackRpc: 'https://cloudflare-eth.com',
+        port: 8545,
+        forkBlockNumber: 18043296n,
+    },
+    {
+        rpcEnv: 'POLYGON_RPC_URL',
+        // Public Polygon RPCs are usually unreliable
+        fallBackRpc: undefined,
+        port: 8137,
+        // Note - this has to be >= highest blockNo used in tests
+        forkBlockNumber: 44215395n,
+    },
+        {
+        rpcEnv: 'FANTOM_RPC_URL',
+        // Public Fantom RPCs are usually unreliable
+        fallBackRpc: undefined,
+        port: 8138,
+        forkBlockNumber: 65313450n,
+    },
+];
+
 export default async function () {
+    for (const network of NETWORKS) {
+        const config = getAnvilOptions(network);
+        await startFork(config);
+    }
+}
```

Comment:
How does this work? Does it start all forks before all tests only once? Or does it restart on every test? I'm trying to get a sense of the impact of this change.
Maybe a better question is: how did this affect test performance? Is it faster/slower than what we had before?

Reply:
It only starts each network once for all tests.
The PR significantly improves test performance for two reasons, 1. Smaller list of pools to fetch onchain state 2. Faster to fetch onchain state locally than over rpc.
There's still a bottleneck with the Subgraph tests.